src/components/Users.jsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from "react";
+import {
+  Container,
+  List,
+  ListItem,
+  ListItemText,
+  TextField,
+} from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import { routes } from "../routes/routes";
+import { searchUsers } from "../utils/utils";
+import { urls } from "../network/urls";
+import { useQuery } from "@tanstack/react-query";
+import { getRequest } from "../network/axiosClient";
+
+const Users = () => {
+
+  const [query, setQuery] = useState("");
+  const [filteredUsers, setFilteredUsers] = useState([])
+
+  const navigate = useNavigate();
+
+  const { data, status } = useQuery({ queryKey: ['users'], queryFn: () => getRequest(urls.USERS) })
```

Comment:
Move this out of rendering and introduce a middle layer which takes input and returns output.